hybrasyl/Objects/Creature.cs

```diff
@@ -908,7 +913,11 @@ public virtual bool UseCastable(Castable castObject, Creature target = null)
                 }
             }
             // Now flood away
-            foreach (var dead in deadMobs) dead.OnDeath();
+            foreach (var dead in deadMobs)
+            {
+                if (dead is Monster) (dead as Monster).OnDeath();
+                else dead.OnDeath();
```

Comment:
this if/else statement does exactly the same thing. There is no need to cast dead as a monster.